src/text-editor-presenter.coffee

```diff
@@ -1253,13 +1253,6 @@ class TextEditorPresenter
     range = marker.getScreenRange()
 
     if decoration.isDestroyed() or not marker.isValid() or range.isEmpty() or not range.intersectsRowRange(@startRow, @endRow - 1)
-      tileStartRow = @tileForRow(range.start.row)
-      tileEndRow = @tileForRow(range.end.row)
-
-      for tile in [tileStartRow..tileEndRow] by @tileSize
-        delete @state.content.tiles[tile]?.highlights[decoration.id]
-
-      @emitDidUpdateState()
```

Comment:
This, as explained in this PR's description, was some leftover code which was useful when we updated highlight state right after an event was triggered. There isn't any un-batched method anymore now and, therefore, we can handle deletion right and only inside `@updateDecorations`. :art: